public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx

```diff
@@ -27,123 +28,137 @@ const RECEIVER_STATUS_POLLING_INTERVAL = 10 * 1000; // 10 seconds
  * 3. (if available) additional metadata about Grafana Managed contact points
  * 4. (if available) the OnCall plugin metadata
  */
-interface UseContactPointsWithStatusOptions {
-  includePoliciesCount: boolean;
-  receiverStatusPollingInterval?: number;
-}
 
-const defaultHookOptions = {
-  includePoliciesCount: true,
-  receiverStatusPollingInterval: RECEIVER_STATUS_POLLING_INTERVAL,
+const {
+  useGetAlertmanagerConfigurationQuery,
+  useGetContactPointsListQuery,
+  useGetContactPointsStatusQuery,
+  useGrafanaNotifiersQuery,
+  useLazyGetAlertmanagerConfigurationQuery,
+  useUpdateAlertmanagerConfigurationMutation,
+} = alertmanagerApi;
+
+const { useGrafanaOnCallIntegrationsQuery } = onCallApi;
+
+/**
+ * Check if OnCall is installed, and fetch the list of integrations if so.
+ *
+ * Otherwise, returns no data
+ */
+const useOnCallIntegrations = ({ skip }: { skip?: boolean } = {}) => {
+  const { installed, loading } = usePluginBridge(SupportedPlugin.OnCall);
+  const oncallIntegrationsResponse = useGrafanaOnCallIntegrationsQuery(undefined, { skip: skip || !installed });
+
+  return useMemo(() => {
+    if (installed) {
+      return oncallIntegrationsResponse;
+    }
+    return {
+      isLoading: loading,
+      data: undefined,
+    };
+  }, [installed, loading, oncallIntegrationsResponse]);
+};
+
+/**
+ * Fetch contact points from separate endpoint (i.e. not the Alertmanager config) and combine with
+ * OnCall integrations and any additional metadata from list of notifiers
+ * (e.g. hydrate with additional names/descriptions)
+ */
+export const useGetContactPoints = () => {
+  const onCallResponse = useOnCallIntegrations();
+  const alertNotifiers = useGrafanaNotifiersQuery();
+  const contactPointsListResponse = useGetContactPointsListQuery();
+
+  return useMemo(() => {
+    const isLoading = onCallResponse.isLoading || alertNotifiers.isLoading || contactPointsListResponse.isLoading;
+
+    if (isLoading || !contactPointsListResponse.data) {
+      return { ...contactPointsListResponse, isLoading: true, contactPoints: [] };
```

Comment:
Are we using `isLoading: true` instead of `isLoading` for some reason?

Reply:
This was the reason for that intermittent "contact point not found" issue.
In this hook, we might have had one of the requests complete, but not the others. But the consumer of the hook would think it's all completed, as `contactPointsListResponse` had finished loading. If we're inside this `if` block, then we should definitely be in a loading state

I'll add a comment to clarify why this is being done :+1: